boards/nxp/imx93_evk/doc/index.rst

```diff
@@ -140,6 +140,28 @@ over dts config. For instance, if ``CONFIG_CAN`` is enabled, MUX A is selected
 even if ``mux="B";`` is configured in dts, and an warning would be reported in
 the log.
 
+User Button GPIO Option
+--------------------------
+
+The user buttons RFU_BTN1 and RFU_BTN2 is connected to i.MX 93 GPIO by default,
+but can be changed to connect to onboard GPIO expander PCAL6524 with on-board DIP
+switches. To do this, switch SW1006 to 0000, then switch SW1005 to 0101. An devicetree
+overlay is included to support this.
+
+Run following command to test user buttons on PCAL6524:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/basic/button
+   :host-os: unix
+   :board: imx93_evk/mimx9352/a55
+   :goals: build
+   :gen-args: -DDTC_OVERLAY_FILE=boards/nxp/imx93_evk/imx93_evk_mimx9352_exp_btn.overlay
```

Comment:
Per my understanding, this overlay is only used for samples/basic/button to test GPIO expander, so I suggest to move it to applications boards directory: samples/basic/button/boards/imx93_evk/imx93_evk_mimx9352_ca55.overlay.

Anyway we could reserve this document, but I suggest move it to be after the chapter "System Reboot (A55)",  we could title it such as "User Buffer GPIO Testing (A55)"

Reply:
There's no other boards that adds overlay to `button` sample, and I suggest not to add this overlay there. Give that:
- This overlay is not mandatory to run the `button`sample
- This overlay is not enabling dependencies for `button` sample

> this overlay is only used for samples/basic/button to test GPIO expander

You can think it that way, but it could be also considered as a "use case" for the board that could be best demostrated using `button` sample. There's no limit if an developer wants to use these buttons on expander on his own application, so we'd better keep it in boards directory.

I know this is not standard, nor is this overlay validated by CI. If you are concerned about that, we could remove the overlay file and just include code blocks in document, just like many other boards did.

Reply:
okay, we can put it in board directory, but move it to boards/nxp/imx93_evk/dts,  and use the following build args in which no need to specify board directory.

`-DEXTRA_DTC_OVERLAY_FILE=imx93_evk_mimx9352_exp_btn.overlay`

Use `EXTRA_DTC_OVERLAY_FILE` but not `DTC_OVERLAY_FILE` as: The build system appends overlays specified in ``EXTRA_DTC_OVERLAY_FILE`` to the overlays in ``DTC_OVERLAY_FILE`` when processing devicetree overlays. This means that changes made via ``EXTRA_DTC_OVERLAY_FILE`` have higher precedence than those made via ``DTC_OVERLAY_FILE``.

Reply:
updated